File: Python/JDI/web/selenium/elements/common/check_box.py
from JDI.core.utils.decorators import scenario
from JDI.web.selenium.elements.base.clickable import Clickable


class CheckBox(Clickable):

    def __init__(self, by_locator=None):
        super(CheckBox, self).__init__(by_locator)

    @scenario("Check element")
    def check(self):
        self.check_action()

    def check_action(self):
        if not self.is_check_action():
            self.click()

    @scenario("Uncheck element")
    def uncheck(self):
       self.uncheck_action()

    def uncheck_action(self):
        if self.is_check_action():
            self.click()

    @scenario(action_name="Get is checked")
    def is_checked(self):
        return self.is_check_action()

    @scenario(action_name="Set value to the element")
    def set_value(self, value):
        self.set_value_action(value)

    def set_value_action(self, value):
        if str(value).lower() in ["true", "check", "1"]:
            self.check()
        elif str(value).lower() in ["false", "uncheck", "0"]:
            self.uncheck()

    def is_check_action(self):
        return self.get_element().is_selected()

File: C#.Net/JDI/Web/JDI Web/Selenium/Elements/Common/CheckBox.cs
using System;
using Epam.JDI.Core.Interfaces.Common;
using JDI_Web.Selenium.Base;
using JDI_Web.Selenium.Elements.Base;
using OpenQA.Selenium;
using static Epam.JDI.Core.Settings.JDISettings;

namespace JDI_Web.Selenium.Elements.Common
{
    public class CheckBox : Clickable, ICheckBox
    {
        public CheckBox() : this(null) { }
        public CheckBox(By byLocator = null, IWebElement webElement = null, WebBaseElement element = null)
            : base(byLocator, webElement, element:element) { }

        public Action<CheckBox> CheckAction = el =>
        {
            if (!el.IsCheckedAction(el))
                el.ClickAction(el);
            if (!el.IsCheckedAction(el))
                throw Exception("Can't check element. Verify locator for click or isCheckedAction");
        };
        public void Check()
        {
            Actions.Check(el => CheckAction(this));
        }

        protected void UncheckAction(WebBaseElement el)
        {
            if (IsCheckedAction((CheckBox)el))
                ClickAction(el);
            if (IsCheckedAction((CheckBox)el))
                throw Exception("Can't uncheck element. Verify locator for click or isCheckedAction");
        }
        public void Uncheck()
        {
            Actions.Uncheck(UncheckAction);
        }

        public Func<CheckBox, bool> IsCheckedAction =
            el => el.IsSelected(el) || el.IsCheckedByAttribute(el);

        public Func<WebBaseElement, bool> IsCheckedByAttribute =
            el => el.WebAvatar.FindImmediately(() => el.WebElement.GetAttribute("checked") != null, false);

        public Func<WebBaseElement, bool> IsSelected =
            el => el.WebAvatar.FindImmediately(() => el.WebElement.Selected, false);

        public bool IsChecked()
        {
            return Actions.IsChecked(el => IsCheckedAction(this));
        }

        protected Action<WebBaseElement, string> SetValueAction = (el, value) =>
        {
            switch (value.ToLower())
            {
                case "true":
                case "1":
                case "check":
                    ((CheckBox) el).Check();
                    break;
                case "false":
                case "0":
                case "uncheck":
                    ((CheckBox) el).Uncheck();
                    break;
                default:
                    throw Exception(
                        $"SetValue not specified correctly {value}, expected: 'true','false','0','1','check','uncheck'");
            }
        };

        public string Value
        {
            get => Actions.GetValue(GetValueFunc);
            set => Actions.SetValue(value, SetValueAction);
        }

        protected Func<WebBaseElement, string> GetValueFunc = el => ((CheckBox) el).IsChecked() + "";

    }
}
